airflow/decorators/base.py

```diff
@@ -275,10 +275,10 @@ def map(
     def partial(
         self, *, dag: Optional["DAG"] = None, task_group: Optional["TaskGroup"] = None, **kwargs
     ) -> "_TaskDecorator[T, OperatorSubclass]":
+        if self.kwargs:
+            raise RuntimeError("Already a partial task")
```

Comment:
Hmmm, this one doesn't behave correctly right now, as:

```
@taks(task_id='x')
def y():
    ...
```

This currently stores `task_id` in the `self.kwargs`

Reply:
Removed this check only in deb54df1c